### DISCUSSION TITLE
bulk_insert and nullable arguments

### DISCUSSION THREAD

**Discussion by lafrech** [sqlalchemy/alembic#1123] [Usage Questions] [answered]
I'm using `bulk_insert` to insert rows with nullable fields.

```py
    op.bulk_insert(
        my_table,
        [
            {
                "name": "Name 1",
            },
            {
                "name": "Name 2",
            },
            {
                "name": "Name 3",
                "optional": "something",
            },
        ],
        # multiinsert=False,
    )
```

I realized that the optional field was not set, silently ignored in fact, unless I used multiinsert.

I don't know if this is intended.

It might be obvious to someone aware of the internals but I found it surprising.

**Comment by zzzeek** [comment, marked as answer]
we could in theory check every entry that the keys are a subset of those keys in the first entry (since that's the one that matters).   We don't have development resources for "nice to haves" right now but feel free to open an issue.
